client_test/helpers_test.go

```diff
@@ -229,3 +231,22 @@ func closeSimulatedChain(t *testing.T, chain chainservice.SimulatedChain) {
 		t.Fatal(err)
 	}
 }
+
+// generateMessageKey generates a ECDSA private key deterministically using the given pk bytes.
+func generateMessageKey(pk []byte) p2pcrypto.PrivKey {
+
+	// We use he given
+	totalSize := 256
+	large := make([]byte, totalSize)
+	copy(large, pk)
+
+	messageKey, _, err := p2pcrypto.GenerateECDSAKeyPair(bytes.NewReader(large))
+	if err != nil {
+		panic(err)
+	}
+
+	if err != nil {
+		panic(err)
+	}
```

Comment:
```suggestion
```

Reply:
Sorry, for some reason I thought something like this would be caught by the linter 🤦